src/zk/zk_client.cc

```diff
@@ -503,6 +503,7 @@ bool ZkClient::GetChildrenUnLocked(const std::string& path, std::vector<std::str
     for (int32_t i = 0; i < data.count; i++) {
         children.push_back(std::string(data.data[i]));
     }
+    deallocate_String_vector(&data);
```

Comment:
how about free in RAII way, in case program accidentally return before e.g in line 504. 
Use `abs::Cleanup`: https://github.com/abseil/abseil-cpp/blob/master/absl/cleanup/cleanup.h#L81

Reply:
ok